ntoskrnl/mm/ARM3/pagfault.c

```diff
@@ -291,16 +291,25 @@ MiCheckVirtualAddress(IN PVOID VirtualAddress,
         else
         {
             /* ReactOS does not supoprt these VADs yet */
-            ASSERT(Vad->u.VadFlags.VadType != VadImageMap);
             ASSERT(Vad->u2.VadFlags2.ExtendableFile == 0);
 
             /* Return the proto VAD */
             *ProtoVad = Vad;
 
-            /* Get the prototype PTE for this page */
-            PointerPte = (((ULONG_PTR)VirtualAddress >> PAGE_SHIFT) - Vad->StartingVpn) + Vad->FirstPrototypePte;
-            ASSERT(PointerPte != NULL);
-            ASSERT(PointerPte <= Vad->LastContiguousPte);
+            /* Check for MEMORY_AREA thingie */
```

Comment:
what is this thingie?

Reply:
the hack where a VAD is a proxy to a MEMORY_AREA from legacy Mm